Automattic/woocommerce-payments
# Avoid saving a session cookie when the currency is changed because of geolocation

Fixes #3761

#### Changes proposed in this Pull Request
As you can see in #3761, some caching systems get invalidated when the option "Automatically switch customers to their local currency if it has been enabled" is enabled because we store that change in the WC session cookie just on the first load without any user interaction.

With this change, `update_selected_currency` function gets a new parameter, `persist_change`, that if `true`, will create a cookie in case it doesn't exist. If the cookie exists, it's updated as normal. If `false`, the cookie is not stored in the user browser if it doesn't exist yet.

This parameter allows us to pass it as `false` in the `update_selected_currency_by_geolocation` function where it's called, making sure that a first-time visitor that has its currency changed via geolocation, doesn't get a cookie in its system until a different action is made (adds a product to the cart, changes manually its currency, or logs in).

#### Testing instructions

* Set the default currency as a different one to the one in your current location
* Go to Payments - Settings - Multi-Currency  and enable `Automatically switch customers to their local currency if it has been enabled`
* Clean the cookies in your browser/open an incognito window
* Visit the store and notice a blue banner in the bottom announcing the change of currency to your local one
* Open the storage info in your browser and see that no cookie has been set
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/2612426/181826634-a452db34-21c3-46f5-b8f8-e2075ce0a7aa.png">
- Make a purchase and make sure that the currency is the one that got changed because of your geolocation.

- Clear cookies again, and enter in the store

- See the notice announcing the change of currency again

- Choose manually a currency different to the one of your location

- See the cookie is created normally to store that change

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/2612426/181827800-54ebd73a-1b95-4a13-8a1f-adef18e8668a.png">


-------------------


**Post merge**
